LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,109 @@ struct OpenAIChatView: View {
+            model: "gpt-3.5-turbo-0613",
+            messages: [
+                Chat(role: .user, content: chat.last?.content ?? ""),
+                Chat(role: .system, content: String(localized: "MULTIPLE_RESOURCE_FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+            ],
+            functions: functions
+        )
+        
+        let result = try await openAI.chats(query: query)
+        
+        var functionCallOutputArray = [String]()
+        
+        let rawFunctionCallOutput = result.choices.rawValue
+        if let rawData = rawFunctionCallOutput.data(using: .utf8) {
+            if let jsonArray = try? JSONSerialization.jsonObject(with: rawData, options: .allowFragments) as? [[String: Any]] {
+                for jsonDictionary in jsonArray {
+                    guard let finishReason = jsonDictionary["finish_reason"] as? String,
+                          let messageDict = jsonDictionary["message"] as? [String: Any],
+                          let functionCallDict = messageDict["function_call"] as? [String: Any],
+                          let arguments = functionCallDict["arguments"] as? String,
+                          let data = arguments.data(using: .utf8),
+                          let argumentsJson = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
+                          let resources = argumentsJson["resources"] as? String else {
+                        print("Error extracting data")
+                        continue
+                    }
+                    
+                    functionCallOutputArray = resources.components(separatedBy: ",")
+                    print("Resources: \(functionCallOutputArray)")
+                }
+            } else {
+                print("Error extracting data")
+            }
+        }
+
+        return functionCallOutputArray
+    }
+
+    private func processFunctionCallOutputArray(functionCallOutputArray: [String], resourcesArray: [FHIRResource]) {
+        var stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        for resource in functionCallOutputArray {
+            var stringResource = resource
+            stringResource = resource.trimmingCharacters(in: .whitespaces)
+            
+            if let index = stringResourcesArray.firstIndex(of: stringResource) {
+                let resourceJSON = resourcesArray[index].jsonDescription
+                print(resourceJSON)
+                chat.append(Chat(role: .system, content: "Based on the function get_resource_title you have requested the health record: \(stringResource). This is the associated JSON data for the resource which you will use to answer the users question: \(resourceJSON). Use this health record to answer the users question ONLY IF the health record is applicable to the question."))
+            } else {
+                print("Resource '\(resource)' not found in stringResourcesArray.")
+            }
+        }
```

Comment:
We will have to adapt the if-statement above (`if viewState == .idle && chat.last?.role == .user`) to also trigger the OpenAI call when the last message is a system message as appended here in the message to ensure the his reaches the OpenAI server: 
```swift
if viewState == .idle && chat.last?.role != .assistant
```